_posts/en/newsletters/2019-03-19-newsletter.md

```diff
@@ -0,0 +1,277 @@
+
+This commitment is inserted into a scriptPubKey template:
+
+<pre>OP_DUP OP_HASH160 OP_PUSH20 <b>6eafa604a503a0bb445ad1f6daa80f162b5605d6</b> OP_EQUALVERIFY OP_CHECKSIG</pre>
+
+Converting the opcodes to hex, this looks like:
+
+    76a9146eafa604a503a0bb445ad1f6daa80f162b5605d688ac
+
+This is inserted into the scriptPubKey part of an output that also
+includes the length of the script (25 bytes) and the amount being paid:
+
+<pre>     amount                           scriptPubKey
+|--------------|  |------------------------------------------------|
+00e1f5050000000019<b>76a9146eafa604a503a0bb445ad1f6daa80f162b5605d688ac</b>
+                |
+    size: 0x19 -> 25 bytes</pre>
+
+This output can then be added to the transaction, which is then signed
+and broadcast.
+
+### Sending to a bech32 address
+
+For an equivalent bech32 P2WPKH address such as
+bc1qd6h6vp99qwstk3z668md42q0zc44vpwkk824zh, you can use one of the
+[reference libraries][bech32 ref libs] to decode the address to a pair
+of two values:
+
+    0 6eafa604a503a0bb445ad1f6daa80f162b5605d6
+
+These two values are also used to create a scriptPubKey template.  The
+first value is the witness script version byte that's used to add a
+value to the stack using one of the opcodes from `OP_0` to `OP_16`.
+The second is the commitment that's also pushed onto the stack:
+
+<pre><b>OP_0</b> OP_PUSH20 <b>6eafa604a503a0bb445ad1f6daa80f162b5605d6</b></pre>
+
+Converting the opcodes to hex, this looks like:
+
+    00146eafa604a503a0bb445ad1f6daa80f162b5605d6
+
+Then, just as before, this is inserted into the scriptPubKey part of an
+output, the output added to the transaction, and the transaction signed
+and broadcast.
+
+For bech32 P2WSH (the segwit equivalent of P2SH) or for future segwit
+witness versions, you don't need to do anything special.  The witness
+script version may be a different number, requiring you to use the
+corresponding `OP_0` to `OP_16` opcode, and the commitment may be a
+different length (up to 40 bytes), but nothing else changes.
+
+What you see above is the entire fundamental change you need to make on
+the backend of your software in order to enable sending to bech32
+addresses.  For most platforms, it should be a very easy change.  See
+[BIP173][] and the [reference implementations][bech32 ref libs] for a
+set of test vectors you can use to ensure your implementation works
+correctly.
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo],
+[LND][lnd repo], [C-Lightning][c-lightning repo], [Eclair][eclair repo],
+[libsecp256k1][libsecp256k1 repo], and [Bitcoin Improvement Proposals
+(BIPs)][bips repo].*
+
+- [LND #2022][] allows the creation of "hold invoices".  These are standard
+  LN invoices that are processed differently when payment is received.
+  Instead of the receiver immediately returning the payment preimage
+  upon receipt of the payment, the receiver delays up until the maximum
```

Comment:
nit: perhaps add 'in order to claim the funds' here, ie:

`immediately returning the payment preimage upon receipt of the payment in order to claim the funds`